lib/Sema/CSDiag.cpp

```diff
@@ -3611,6 +3611,19 @@ static bool tryRawRepresentableFixIts(InFlightDiagnostic &diag,
     } else if (!fromTypeIsOptional) {
       diag.fixItInsert(exprRange.Start, convWrapBefore);
       diag.fixItInsertAfter(exprRange.End, convWrapAfter);
+    } else {
+      SmallString<16> fixItBefore(convWrapBefore);
+      SmallString<16> fixItAfter;
+
+      if (expr->needsParensInsidePostfixOperator()) {
+        fixItBefore += "(";
+        fixItAfter = ")";
+      }
+
+      fixItAfter += "!" + convWrapAfter.str();
```

Comment:
It's dangerous to suggest a {{!}}, even though for NS_STRING_ENUM cases it would be okay. Maybe we should put the fix-it on a note in this case?

```c++
diag.flush();
diagnose(…)
  .fixItInsert(…)
  .fixItInsert(…);
```